core/src/main/java/org/apache/calcite/rex/RexCopier.java

```diff
@@ -52,11 +53,30 @@ private RelDataType copy(RelDataType type) {
   }
 
   public RexNode visitOver(RexOver over) {
-    throw new UnsupportedOperationException();
+    final boolean[] update = null;
+    return new RexOver(copy(over.getType()), over.getAggOperator(),
+        visitList(over.getOperands(), update), visitWindow(over.getWindow()),
+        over.isDistinct(), over.ignoreNulls());
   }
 
   public RexWindow visitWindow(RexWindow window) {
```

Comment:
The method visitOver in the superclass seems to do exactly the same thing. Can you clarify what's the difference?

Reply:
The difference is that in this method the type of RexOver is ` copy(over.getType())`  instead of `over.getType()`.

Reply:
I'm sorry @chunweilei. My previous comment was actually for visitWindow which does not seem to involve a copy of a type.

Reply:
To make sure that it will always return a new `RexWindow`. I think it is necessary if `RexCopier` is a deep copy.

Reply:
I would assume that if every other visit method creates a new instance of a RexNode then RexShuttle would take care of it. Sorry for insisting on this but I would prefer if don't we duplicate pieces of code (especially between parent-child classes). I will give it some thought ;)

Reply:
I believe you are right. But I would like to spend some time confirming it. Really appreciate your careful attention to detail.

Reply:
I guess you could remove entirely the method, instead of calling super.

Reply:
Yes, we can. I keep it on purpose to tell we don't forget this method. I will remove it since it seems redundant.